src/ai/backend/common/api_handlers.py

```diff
@@ -0,0 +1,323 @@
+                f"Parameter '{input_param_type}' must use one of QueryParam, PathParam, HeaderParam, MiddlewareParam, BodyParam"
+            )
+
+    except ValidationError as e:
+        raise InvalidAPIParameters(str(e))
+
+
+class _HandlerParameters:
+    _params: dict[str, _ParamType]
+
+    def __init__(self) -> None:
+        self._params: dict[str, _ParamType] = {}
+
+    def add(self, name: str, value: _ParamType) -> None:
+        if value is not None:
+            self._params[name] = value
+
+    def get_all(self) -> dict[str, _ParamType]:
+        return self._params
+
+
+async def _parse_and_execute_handler(request: web.Request, handler, signature) -> web.Response:
+    handler_params = _HandlerParameters()
+    for name, param in signature.parameters.items():
+        # If handler has no parameter, for loop is skipped
+        # Raise error when parameter exists and has no type hint or not wrapped by 'Annotated'
+        if param.annotation is inspect.Parameter.empty:
+            raise InvalidAPIParameters(
+                f"Type hint or Annotated must be added in API handler signature: {param.name}"
+            )
+
+        value = await _extract_param_value(request=request, input_param_type=param.annotation)
+
+        if not value:
+            raise InvalidAPIParameters(
+                f"Type hint or Annotated must be added in API handler signature: {param.name}"
+            )
+
+        handler_params.add(name, value)
+
+    response = await handler(**handler_params.get_all())
+
+    if not isinstance(response, APIResponse):
+        raise InvalidAPIParameters(
+            f"Only Response wrapped by APIResponse Class can be handle: {type(response)}"
+        )
+
+    return web.json_response(
+        response.to_json,
+        status=response._status_code,
+    )
+
+
+def api_handler(handler):
+    """
+    This decorator processes HTTP request parameters using Pydantic models.
+
+    1. Request Body:
+        @api_handler
+        async def handler(body: BodyParam[UserModel]):  # UserModel is a Pydantic model
+            user = body.parsed                          # 'parsed' property gets pydantic model you defined
+            # Response model should inherit BaseResponseModel
+            return APIResponse.build(status_code=200, response_model=YourResponseModel(user=user.id))
+
+    2. Query Parameters:
+        @api_handler
+        async def handler(query: QueryParam[QueryPathModel]):
+            parsed_query = query.parsed
+            return APIResponse.build(status_code=200, response_model=YourResponseModel(search=parsed_query.query))
+
+    3. Headers:
+        @api_handler
+        async def handler(headers: HeaderParam[HeaderModel]):
+            parsed_header = headers.parsed
+            return APIResponse.build(status_code=200, response_model=YourResponseModel(data=parsed_header.token))
+
+    4. Path Parameters:
+        @api_handler
+        async def handler(path: PathModel = PathParam(PathModel)):
```

Comment:
```suggestion
        async def handler(path: PathParam[PathModel]):
```
I think this needs a correction 👀

Reply:
Yes, It should be fixed